zboxcore/sdk/uploadworker.go

```diff
@@ -618,6 +619,219 @@ func (req *UploadRequest) processUpload(ctx context.Context, a *Allocation) {
 	return
 }
 
+func (req *UploadRequest) processStreamingUpload(ctx context.Context, a *Allocation) {
```

Comment:
It would be great if we had a unit test for this new processStreamingUpload method.